src/EnergyPlus/SystemAvailabilityManager.cc

```diff
@@ -491,7 +493,9 @@ namespace SystemAvailabilityManager {
                                                                          lAlphaFieldBlanks,
                                                                          cAlphaFieldNames,
                                                                          cNumericFieldNames);
-                if ((ErrorsFound = cAlphaArgs(1).empty())) ShowSevereError(state, format("{} Name, cannot be blank", cCurrentModuleObject));
+
+                UtilityRoutines::IsNameEmpty(state, cAlphaArgs(1), cCurrentModuleObject, ErrorsFound);
+
```

Comment:
Yes, I messed it up, thanks for the catch.

Reply:
Would you like me to add a unit test to this? @mbadams5

Reply:
I don't think this needs a unit test, and to be honest, this code isn't even really needed... The InputProcessor doesn't allow blank names for these objects per the epJSON Schema, so the initial InputProcessor validation would catch this error before this point.

Reply:
Got it :)

Reply:
Actually, I messed up. I should have said `ErrorsFound |=`, not `ErrorsFound =`. But anyways, I'm glad we're back to the safety of tried and true patterns from 1995 where we are calling functions and constructing and passing parameters for conditions that only occur 0.000001% of the time. I mean, we do need to leave some work for EnergyPlus 10X part 2: Electric Boogaloo. 🙄

Reply:
Sorry for the snark, it's been a bad week/month/six months. I understand we cannot boil the ocean here, much as I would like to. Thanks for all of your help with this and everything else @mbadams5. Are there really five other mbadams on GitHub? Jeez.

Reply:
@amirroth, I am perfectly fine to edit this PR to make it not use an existing function. It was obviously a very quick fix and not trying to make a major refactor.

I could also change the code to be the following as well:
```c++
if (cAlphaArgs(1).empty()) {
    ShowSevereError(state, format("{} Name, cannot be blank", cCurrentModuleObject));
    ErrorsFound = true;
}
```

I commented in the original PR, but on your `ErrorsFound |=` point, it won't necessarily work here.

If `ErrorsFound` is already `true` before this if statement, then it will output the SevereError regardless of whether or not the name field is blank.

Since the same `ErrorsFound` variable is used for ALL SystemAvailability types input processing, there is a possibility of the above happening.

Finally, like I already mentioned, we really don't need to be doing this validation regardless. The InputProcessor already validates that the name is not blank (similar to the choice fields and enums)...

Reply:
So just delete it and start the trend. What does the json error look like when this field is blank?

5ZoneWarmest: IDFEditor used to delete name. Doesn't show me which avail manager has the issue but for sure the severe says there is a problem before get input is called.

    ** Severe  ** <root>[AvailabilityManager:Scheduled] - Object contains a property that could not be validated using 'properties' or 'additionalProperties' constraints: ''.
    ** Severe  ** <root>[AvailabilityManager:Scheduled] - Object name is required and cannot be blank or whitespace
    **  Fatal  ** Errors occurred on processing input file. Preceding condition(s) cause termination.

Reply:
💯